server/index.js

```diff
@@ -41,6 +58,123 @@ app.post('/userEventData', (req, res) => {
   res.sendStatus(201);
 });
 
+const buildUserProfilesMsg = (msgObj) => {
+  return {
+    userId: msgObj.userId,
+    groupId: msgObj.groupId,
+    events: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 ?
+        prev.concat({
+          movie: {
+            id: curr.movieObj.id,
+            profile: curr.movieObj.profile,
+          },
+          progress: curr.progress,
+          startTime: new Date(curr.time - (6000000 * curr.progress)),
+        })
+        : prev;
+    }, []),
+  };
+};
+
+const buildRecommendationMsg = (msgObj) => {
+  return {
+    userId: msgObj.userId,
+    groupId: msgObj.groupId,
+    recs: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 && curr.movieObj.isRec === true && curr.progress === 1 ?
+        prev + curr.progress : prev;
+    }, 0),
+    nonRecs: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 && curr.movieObj.isRec === false && curr.progress === 1 ?
+        prev + curr.progress : prev;
+    }, 0),
+  };
+};
+
+const buildAppServerMsg = (msgObj) => {
+  return {
+    userId: msgObj.userId,
+    events: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 ?
+        prev.concat({
+          movie: { id: curr.movieObj.id },
+          progress: curr.progress,
+          startTime: new Date(curr.time - (6000000 * curr.progress)),
+        })
+        : prev;
+    }, []),
+  };
+};
+
+let eventDashboard = [];
+app.post('/newEvent', (req, res) => {
+  db.addEvent(req.body.session, req.body.query)
+    .then((msgObj) => {
```

Comment:
Lines 113-142 can all be inside one then() body, what is the reason behind chaining them?